src/Controller/MainController.php

```diff
@@ -37,13 +37,35 @@ public function __construct(\Twig_Environment $twig, EntityRepository $threadRep
      */
     public function indexAction(Request $request)
     {
-        $response = new Response();
-        $threads = $this->threadRepository->findAll();
+
+
+        //$allThreads = count( $this->threadRepository -> findAll());
+        $currentPage = $request->get('page', 1);
+
+        //var_dump(($currentPage - 1) * 10); die;
+
+        $totalCount = $this->threadRepository->createQueryBuilder('u')
+            ->select('count(u.id)')
+            ->getQuery()
+            ->getSingleScalarResult();
+
+        $threads = $this->threadRepository
+            ->createQueryBuilder('t')
+            ->setFirstResult(($currentPage-1)*10)
+            ->setMaxResults(10)
+            ->getQuery()
+            ->execute();
+        ;
+
+        $totalPages = ceil($totalCount/ 10);
```

Comment:
could be worth putting the number 10 in a variable like $numberOfResultsPerPage so if you need to change it there is only one place.